### PR TITLE
feat: allow plugin access to current peer

### PR DESCRIPTION
**Pull request by alexbarnsley**
Give plugins the ability to access the current core peer.

## What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactor
- [ ] Performance
- [ ] Tests
- [ ] Build
- [ ] Documentation
- [ ] Code style update
- [ ] Continuous Integration
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Does this PR release a new version?

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch
- [x] All tests are passing
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/renderer/services/plugin-manager.js [hunk 1]
```diff
@@ -620,6 +620,14 @@ class PluginManager {
       sandbox.walletApi.http = new PluginHttp(config.urls)
     }
 
+    if (config.permissions.includes('PEER_CURRENT')) {
+      sandbox.walletApi.peers = {
+        getCurrent: () => {
+          return cloneDeep(this.app.$client.client)
```
Comment (luciorubeens) [inline src/renderer/services/plugin-manager.js]: It's exposing the `typescript-client` instance, is that correct?

I can make any request just by changing the host:

```
const client = walletApi.peers.getCurrent();
client.host = 'https://api.github.com'
const response = await client.get('orgs/arkecosystem')
```

Comment (alexbarnsley) [reply src/renderer/services/plugin-manager.js]: Hm did that work for you? `host` is supposed to be [private readonly](https://github.com/ArkEcosystem/typescript-client/blob/master/src/connection.ts#L10). I'll take a look and find a way around it, thanks 👌

Comment (alexbarnsley) [reply src/renderer/services/plugin-manager.js]: I guess it's because it's converted to JS, the private readonly is disregarded

**Comment by alexbarnsley** [comment]
@luciorubeens could you have a gander, please. I've made it more restrictive

**Pull request merged**